src/molecules/Site.h

```diff
@@ -51,16 +51,17 @@ class Site {
 	   \endcode
 	 */
 	virtual void readXML(XMLfileUnits& xmlconfig) {
-		xmlconfig.getNodeValueReduced("coords/x", _r[0]);
-		xmlconfig.getNodeValueReduced("coords/y", _r[1]);
-		xmlconfig.getNodeValueReduced("coords/z", _r[2]);
-		xmlconfig.getNodeValueReduced("mass", _m);
-
 		if (!xmlconfig.getNodeValue("@name", _name)) {
```

Comment:
The names are not read in correctly as this is (weirdly) always true. I think that getNodeValue returns the number of found nodes and not a boolean value as often expected.

Reply:
This is probably a bug which also occurs in other parts of the code...

Reply:
The present bug could be fixed with something like

```suggestion
		// Read name; use type as name if name is not defined
		std::string name = "";
		xmlconfig.getNodeValue("@name", name);
		if (name.empty()) {
			Log::global_log->error() << "Cannot find site name. Defaulting to type." << std::endl;
			xmlconfig.getNodeValue("@type", _name);
		} else {
			_name = name;
		}
```

Reply:
We can discuss if a warning would be better than an error here as almost all (legacy) components.xml files do not include a name. The name functionality was added together with the ADIOS support about 3 years ago, if I remember correctly.

Reply:
Yes, I was also thinking of changing this to a warning - but I wonder if site names are important and were used at all since they were introduced in 995c8f22949076a58149ee240286c087b76cbf80. I cannot find any reference for this but it might be used for visualizations.
However, I wanted to keep this PR short and limited in scope not touching anything existing.

Reply:
If I remember correctly, the idea was to use the names in the visualization with some kind of automatism for the selection of the [correct color](https://en.wikipedia.org/wiki/CPK_coloring).

Reply:
OK. Sounds like it is useful and should be kept then. Will change it to a warning message for now as it does not affect the simulation itself.

Reply:
> The names are not read in correctly as this is (weirdly) always true. I think that getNodeValue returns the number of found nodes and not a boolean value as often expected.

Seems the XML parser does not handle attributes correctly here. I opened issue #310  to track this.